FIX: generate the proper app event when changing user status

AndrewPrigorshnev opened the pull request:
<!-- NOTE: All pull requests should have tests (rspec in Ruby, qunit in JavaScript). If your code does not include test coverage, please include an explanation of why it was omitted. -->

tgxworld commented:
It feels like we have a missing test here to ensure that the `site-header` component which is the one listening to the `user-status:changed` event has re-rendered.

AndrewPrigorshnev commented:
@tgxworld we have an acceptance test that test in on a higher level:

https://github.com/discourse/discourse/blob/99319142035d4c4c1b340dcde324454346cab91a/app/assets/javascripts/discourse/tests/acceptance/user-status-test.js#L77

That test make sure that after setting the status, the header gets updated. There is also the same tests for updating and clearing the status.

But the question arises why that test didn't fail when I was sending the wrong event 😄 . That was because firing the wrong `do-not-disturb:changed` event also triggers re-rendering of the header.

I don't really want to add one more test for this implementation detail (that it's the `user-status:changed` should be triggering rerendering).

The header must get updated after changing the status, this is what matters. Even though those tests didn't catch the typo, we may rely on them. They cover the desired behaviour, they will fail in case of problems with updating the header.

AndrewPrigorshnev commented:
I've just realized we don't need to merge this PR anymore. I moved event triggering in another PR to the new place and I trigger the right event there:

https://github.com/discourse/discourse/blob/943cae82da88fe7e6b9d436162ff614ba8a75836/app/assets/javascripts/discourse/app/models/user.js#L1005-L1008

So, we probably need to close this PR. But let's finish discussing tests, if you feel like we need one more test case here.

tgxworld commented:
> The header must get updated after changing the status, this is what matters. Even though those tests didn't catch the typo, we may rely on them.

I agree here. How we expect the app to behave is all that matters although it is nice that we caught this very specific typo too.
